### PR TITLE
Fix network switching issues with WC

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes [RNBW-2077](https://linear.app/rainbow/issue/RNBW-2077/wrong-chain-showed-in-connected-apps-after-chain-switching-is)

Dapps that ask for a wallet to switch chains are successful, but the change is not displayed visually in our "Connected Apps" sheet (we always show them connected to Ethereum Mainnet). When users initiate a transaction, they see the expected chain.

_Note: this shouldn't be confused with chain switching **within** Rainbow_

## What changed (plus any additional context for devs)

- `chainId`s need to be converted from strings to numbers for `ethereumUtils.getNetworkFromChainId`. We do this in some places, but not others.
- wallet connection changes needed to be persisted to redux when listening for wallet connect chain adding/switching events, this is important for dapps on L2s to reflect correct chain in our "connected apps".

## PoW (screenshots / screen recordings)

Video of broken version on current `develop` branch

https://user-images.githubusercontent.com/3009331/146468794-d1a6aeda-1e40-4b54-84e7-80aa02648fcc.mp4

Video of fixed version on `@bruno/fix-wc-network-display`

https://user-images.githubusercontent.com/3009331/146475133-956ecffb-1bcc-4150-96d4-1bd2a5504221.mp4

## Dev checklist for QA: what to test

1. Connect to L2 dapp like Hop (hop.exchange, see above video.)
2. Validate that you are connected to Mainnet
3. Initiate a transaction from L2 --> L1, i.e. Arbitrum to Mainnet.
4. Tap "Send" (in Hop)
5. Go to Rainbow, confirm chain switching. Message should show "Arbitrum One" in text.
6. Check your Connected Apps:
  - Expected: Hop will be connected to "Arbitrum"
  - Actual: Hop is connected to "Mainnet"
7. Go back to Hop. Tap "Send" again to bridge transfer.
8. Go back to Rainbow. See that the transaction is using the Expected Chain (meaning that only our Connected Apps list needs to be updated)

## Final checklist
- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/redux/walletconnect.js [hunk 1]
```diff
@@ -549,22 +550,6 @@ export const removeWalletConnector = peerId => (dispatch, getState) => {
...
-    const connector = walletConnectors[key];
-    const newSessionData = {
-      accounts: [accountAddress],
-      chainId,
-    };
-    connector.updateSession(newSessionData);
-
-    saveWalletConnectSession(connector.peerId, connector.session);
-  });
-};
```
Comment (nickbytes) [inline src/redux/walletconnect.js:566]: Exported but never used.

File: src/App.js [hunk 2]
```diff
@@ -254,7 +254,7 @@ class App extends Component {
     Navigation.setTopLevelNavigator(navigatorRef);
 
   handleTransactionConfirmed = tx => {
-    const network = ethereumUtils.getNetworkFromChainId(tx.chainId);
```
Comment (nickbytes) [inline src/App.js:257]: @brunobar79 should this change be separate from networking switching issues with WC?

Comment (brunobar79) [reply src/App.js:257]: ideally yes but not worth the pain of pulling it out

**Review by estebanmino — APPROVED** [review]
code LGTM

**Comment by estebanmino** [comment]
curious to know if we can change the new connection request network 
image

for ex. there we have ethereum but is an arbitrum connection request

**Comment by nickbytes** [comment]
@estebanmino – @brunobar79 and I talked about this being confusing too. Feels like the "Arbitrum One" in the text needs to stand out more. And the network in the bottom right needs to say something like "Current Network".

What do you think @christianbaroni ?

**Comment by brunobar79** [comment]
@nickbytes WDYT if we get this merged (since it fixed an actual bug) and then we can improve UI/UX separately?

**Comment by nickbytes** [comment]
Yep definitely not a blocker.

**Review by nickbytes — APPROVED** [review]

**Pull request merged**